src/components/recovery/RecoveryStatus/index.tsx

```diff
@@ -1,12 +1,34 @@
 import { SvgIcon, Typography } from '@mui/material'
+import { useContext } from 'react'
 import type { ReactElement } from 'react'
 
 import ClockIcon from '@/public/images/common/clock.svg'
 import { useRecoveryTxState } from '@/hooks/useRecoveryTxState'
 import type { RecoveryQueueItem } from '@/services/recovery/recovery-state'
+import { PendingStatus } from '@/store/pendingTxsSlice'
+import { RecoveryContext } from '../RecoveryContext'
+
+const STATUS_LABELS: Partial<Record<PendingStatus, string>> = {
+  [PendingStatus.SUBMITTING]: 'Submitting',
```

Comment:
Should we use the `RecoveryEvent` instead? It doesn't have the Submitting event though (not sure if there is an equivalent)

Reply:
As discussed: this will be left as is as it is not event but pending status related.